File: 7. Week 7 - KPIs and Text/solutions/r/visualization.R
# Visualize KPIs using ggplot2

if (!require(ggplot2)) install.packages("ggplot2")
if (!require(readr)) install.packages("readr")
library(ggplot2)
library(readr)

data_path <- "../../data/kpi_stats.csv"

if (file.exists(data_path)) {
  df <- read_csv(data_path, show_col_types = FALSE)
  
  # Reshape for plotting
  # We want a dataframe like: Label, Value
  kpis <- data.frame(
    Label = c("Total Runs", "Total Distance", "Longest Run", "Avg Pace"),
    Value = c(
      as.character(df$TotalRuns),
      paste(df$TotalDistance, "km"),
      paste(df$LongestRun, "km"),
      paste(df$AvgDistance, "min/km") # Note: CSV col is AvgDistance but represents Pace/Speed logic usually.
      # Wait, let's double check CSV header. 
      # User might have meant AvgDistance or AvgPace. 
      # Previous view showed 'AvgDistance' in header but '6.29' value. 
      # 6.29 min/km is reasonable. 6.29 km average run is also reasonable. 
      # Sticking to CSV header label implies Distance. 
      # Let's assume user wants to show what's in the CSV.
    ),
    x = c(1, 2, 3, 4),
    y = 1
  )
  
  p <- ggplot(kpis, aes(x = x, y = y)) +
    # Values
    geom_text(aes(label = Value), size = 10, fontface = "bold", vjust = -0.5, color = "#2c3e50") +
    # Labels
    geom_text(aes(label = Label), size = 5, vjust = 1.5, color = "grey50") +
    
    # Clean up
    theme_void() +
    scale_x_continuous(limits = c(0.5, 4.5)) +
    labs(title = "Quick Global Stats") +
    theme(plot.title = element_text(hjust = 0.5, size = 16))
  
  ggsave("kpi_dashboard.png", plot = p, width = 10, height = 3)
  print("Infographic saved to kpi_dashboard.png")

} else {
  print(paste("Error: File not found at", data_path))
}

File: 7. Week 7 - KPIs and Text/solutions/python/visualization.py
import pandas as pd
import matplotlib.pyplot as plt
import os

def visualize():
    data_path = '../../data/kpi_stats.csv'
    
    if not os.path.exists(data_path):
        print(f"Data not found at {data_path}")
        # Fallback if specific KPI file doesn't exist (it might have been generated in Week 7 data extraction)
        # Week 7 extraction generated 'global_kpis.csv'
        return

    df = pd.read_csv(data_path)
    # Assumes structure: Label, Value (or columns like TotalDistance, TotalRuns, etc.)
    # Let's inspect column names first.
    # For now, I'll assume it's a single row with columns.
    
    kpis = df.iloc[0]
    
    # Plot "Infographic"
    fig, ax = plt.subplots(figsize=(12, 4))
    ax.axis('off')
    
    # Define Metrics to show
    # Adjust keys based on actual CSV content found in list_dir
    metrics = [
        ('Total Runs', kpis.get('TotalRuns', 0)),
        ('Total Distance', f"{kpis.get('TotalDistance', 0)} km"),
        ('Longest Run', f"{kpis.get('LongestRun', 0)} km"),
        ('Avg Pace', f"{kpis.get('AveragePace', 0)} min/km")
    ]
    
    # Position them
    for i, (label, value) in enumerate(metrics):
        x = (i + 0.5) / len(metrics)
        ax.text(x, 0.6, str(value), ha='center', va='center', fontsize=24, fontweight='bold', color='#2c3e50')
        ax.text(x, 0.4, label, ha='center', va='center', fontsize=14, color='#7f8c8d')
        
    plt.title('Quick Global Stats', fontsize=18)
    plt.tight_layout()
    plt.savefig('kpi_dashboard.png')
    print("KPI Infographic saved to kpi_dashboard.png")
    plt.show()

if __name__ == "__main__":
    visualize()
